modules/yugabytedb/yugabytedb.go

```diff
@@ -0,0 +1,246 @@
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, fmt.Errorf("customize: %w", err)
+		}
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *YugabyteDBContainer
+	if container != nil {
+		c = &YugabyteDBContainer{
+			Container:            container,
+			ysqlDatabaseName:     req.Env[ysqlDatabaseNameEnv],
+			ysqlDatabaseUser:     req.Env[ysqlDatabaseUserEnv],
+			ysqlDatabasePassword: req.Env[ysqlDatabasePasswordEnv],
+		}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("generic container: %w", err)
+	}
+
+	return c, nil
+}
+
+// YSQLConnectionString returns a connection string for the yugabyteDB container
+// using the configured database name, user, password, port, host and additional
+// arguments.
+// Additional arguments are appended to the connection string as query parameters
+// in the form of key=value pairs separated by "&".
+//
+// # Example:
+//
+//	connStr, err := y.YSQLConnectionString(ctx, "sslmode=disable")
+//	if err != nil {
+//		return err
+//	}
+//
+// The above code will return a connection string with the sslmode set to disable.
+func (y *YugabyteDBContainer) YSQLConnectionString(ctx context.Context, args ...string) (string, error) {
+	host, err := y.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	mappedPort, err := y.MappedPort(ctx, ysqlPort)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf(
+		"postgres://%s:%s@%s/%s?%s",
+		y.ysqlDatabaseUser,
+		y.ysqlDatabasePassword,
+		net.JoinHostPort(host, mappedPort.Port()),
+		y.ysqlDatabaseName,
+		strings.Join(args, "&"),
+	), nil
+}
+
+// YCQLConfigureClusterConfig merges the provider [gocql.ClusterConfig] with the
+// yugabyteDB container configuration.
+// The method sets the hosts, keyspace and authenticator fields of the cluster
+// configuration to the values of the yugabyteDB container.
+//
+// # Example:
+//
+//	cluster := gocql.NewCluster()
+//	ctr.YCQLConfigureClusterConfig(ctx, cluster)
+//	session, err := cluster.CreateSession()
+//	if err != nil {
+//		return err
+//	}
+//	defer session.Close()
+//
+// [gocql.ClusterConfig]: https://pkg.go.dev/github.com/gocql/gocql?tab=doc#ClusterConfig
+func (y *YugabyteDBContainer) YCQLConfigureClusterConfig(ctx context.Context, cfg *gocql.ClusterConfig) error {
+	host, err := y.Host(ctx)
+	if err != nil {
+		return err
+	}
+
+	mappedPort, err := y.MappedPort(ctx, ycqlPort)
+	if err != nil {
+		return err
+	}
+
+	inspect, err := y.Container.Inspect(ctx)
+	if err != nil {
+		return err
+	}
+
+	var (
+		keyspace string
+		user     string
+		password string
+	)
+
+	for _, env := range inspect.Config.Env {
+		parts := strings.SplitN(env, "=", 2)
+		switch parts[0] {
+		case ycqlKeyspaceEnv:
+			keyspace = parts[1]
+		case ycqlUserNameEnv:
+			user = parts[1]
+		case ycqlPasswordEnv:
+			password = parts[1]
+		}
```

Comment:
suggestion: use [strings.Cut](https://pkg.go.dev/strings#Cut) to validate, SplitN can return nil which would fail.
```suggestion
		key, val, found := strings.Cut(env, "=")
		if !found {
			continue
		}
		switch key {
		case ycqlKeyspaceEnv:
			keyspace = val
		case ycqlUserNameEnv:
			user = val
		case ycqlPasswordEnv:
			password = val
		}
```